### PR TITLE
TCP Transport - client-related code moved to own struct

### PR DESCRIPTION
**Pull request by electricmonk**
This PR moves all code that deals with a single client connection within `DirectTransport` to a new struct, `clientConnection`. This is in preparation for a future PR to allow updating peer topology in runtime.

### CODE REVIEW

File: services/gossip/adapter/tcp/direct_transport.go [hunk 1]
```diff
@@ -112,31 +111,26 @@ func newDirectTransportServer() *lockableTransportServer {
...
 func (t *DirectTransport) connectForever(bgCtx context.Context, peerNodeAddress string, peer config.GossipPeer) {
-	t.outgoingQueues.Lock()
-	defer t.outgoingQueues.Unlock()
+	t.clientConnections.Lock()
+	defer t.clientConnections.Unlock()
 
 	if peerNodeAddress != t.config.NodeAddress().KeyForMap() {
 
-		newQueue := NewTransportQueue(SEND_QUEUE_MAX_BYTES, SEND_QUEUE_MAX_MESSAGES, t.metricRegistry, peerNodeAddress)
+		client := newClientConnection(peer.HexAddress()[:6], peer, t.logger, t.metricRegistry, t.metrics, t.config)
```
Comment (ronnno) [inline services/gossip/adapter/tcp/direct_transport.go]: why do you enforce the length of HexAddress here?

File: services/gossip/adapter/tcp/direct_outgoing.go [hunk 2]
```diff
@@ -10,56 +10,107 @@ import (
...
 		}
 	}
 }
 
 // returns true if should attempt reconnect on error
-func (t *DirectTransport) clientHandleOutgoingConnection(ctx context.Context, conn net.Conn, queue *transportQueue) bool {
-	t.logger.Info("successful outgoing gossip transport connection", log.String("peer", queue.networkAddress), trace.LogFieldFrom(ctx))
-	t.metrics.activeOutgoingConnections.Inc()
-	defer t.metrics.activeOutgoingConnections.Dec()
+func (c *clientConnection) clientHandleOutgoingConnection(ctx context.Context, conn net.Conn, queue *transportQueue) bool {
```
Comment (ronnno) [inline services/gossip/adapter/tcp/direct_outgoing.go]: suggest to extract some of the branching code here into functions - replace comments with function names to make this easier to read

File: config/config.go [hunk 3]
```diff
@@ -233,6 +233,7 @@ type ValidatorNode interface {
 type GossipPeer interface {
 	GossipPort() int
 	GossipEndpoint() string
+	HexAddress() string
```
[the remark this answers is not in the document]
Comment (electricmonk) [reply config/config.go]: Orbs address. Id for logs mostly.

File: services/gossip/adapter/tcp/direct_transport.go (hunk 1 above)
Comment (electricmonk) [reply services/gossip/adapter/tcp/direct_transport.go]: I do not enforce the length - I use only the first 6 chars

**Comment by electricmonk** [comment]
good review; handled most of it in another commit
re connection object - I think it's actually beautiful that we don't need to reference it; it's cleaner, IMO, maybe more functional, that we don't "pollute the state" of the client with anything mutable.

**Pull request merged**